crypto/dh/dh_key.c

```diff
@@ -230,3 +230,33 @@ static int dh_finish(DH *dh)
     BN_MONT_CTX_free(dh->method_mont_p);
     return 1;
 }
+
+int DH_buf2key(DH *dh, const unsigned char *buf, size_t len)
+{
+    BIGNUM *pubkey;
+    if ((pubkey = BN_bin2bn(buf, len, NULL)) == NULL
+        || BN_is_zero(pubkey)
+        || !DH_set0_key(dh, pubkey, NULL))
+        goto err;
+    return 1;
+err:
+    DHerr(DH_F_BUF2KEY, ERR_R_INTERNAL_ERROR);
+    BN_free(pubkey);
+    return 0;
+}
+
+size_t DH_key2buf(const DH *dh, unsigned char **pbuf)
+{
+    const BIGNUM *pubkey;
+    DH_get0_key(dh, &pubkey, NULL);
+    if (pubkey == NULL) {
+        DHerr(DH_F_KEY2BUF, ERR_R_PASSED_INVALID_ARGUMENT);
```

Comment:
The travis red cross is relevant and is related to this and similar lines. The "F" code here doesn't match the function name.